src/plots/gl2d/camera.js

```diff
@@ -76,32 +99,114 @@ function createCamera(scene) {
                             (viewBox[3] - viewBox[1]) * (dataBox[3] - dataBox[1]) +
                         dataBox[1];
 
-                    if(!result.boxEnabled) {
+                    if(!result.boxInited) {
                         result.boxStart[0] = dataX;
                         result.boxStart[1] = dataY;
+                        result.dragStart[0] = x;
+                        result.dragStart[1] = y;
                     }
 
                     result.boxEnd[0] = dataX;
                     result.boxEnd[1] = dataY;
 
-                    result.boxEnabled = true;
+                    // we need to mark the box as initialized right away
+                    // so that we can tell the start and end pionts apart
+                    result.boxInited = true;
+
+                    // but don't actually enable the box until the cursor moves
+                    if(!result.boxEnabled && (
+                        result.boxStart[0] !== result.boxEnd[0] ||
+                        result.boxStart[1] !== result.boxEnd[1])
+                    ) {
+                        result.boxEnabled = true;
+                    }
+
+                    // constrain aspect ratio if the axes require it
+                    var smallDx = Math.abs(result.dragStart[0] - x) < MINZOOM;
+                    var smallDy = Math.abs(result.dragStart[1] - y) < MINZOOM;
+                    if(getSubplotConstraint() && !(smallDx && smallDy)) {
+                        dx = result.boxEnd[0] - result.boxStart[0];
+                        dy = result.boxEnd[1] - result.boxStart[1];
+                        var dydx = (dataBox[3] - dataBox[1]) / (dataBox[2] - dataBox[0]);
+
+                        if(Math.abs(dx * dydx) > Math.abs(dy)) {
+                            result.boxEnd[1] = result.boxStart[1] +
+                                Math.abs(dx) * dydx * (Math.sign(dy) || 1);
+
+                            // gl-select-box clips to the plot area bounds,
+                            // which breaks the axis constraint, so don't allow
+                            // this box to go out of bounds
```

Comment:
in contrast, svg lets you make these constrained boxes extend outside the plot area. Didn't seem worth diving into `gl-select-box` to make this marginal feature match up precisely.